app/models/session.rb

```diff
@@ -5,6 +5,13 @@ class Session < ApplicationRecord
   scope :alive, -> { where alive? }
   scope :expired, -> { where expired? }
 
+  def active?(providers = {})
+    return true unless self.expires_at
+    return false unless self.expires_at > Time.now.utc
+    return true if providers.empty?
+    providers.any? { |provider| self.data['provider'] == provider }
```

Comment:
I forgot the syntax meant it was an optional argument. Right now, since there's only one provider, the optional arg is the way to go. Thanks.